### PR TITLE
Move ITR skipping logic to test events handler

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
# What Does This Do
Logic that is responsible for CI Visibility ITR skipping is moved from individual test framework instrumentations to `TestEventHandler` class.

# Motivation
Getting rid of duplicate code, and preparing for further changes in ITR implementation (list of skippable tests will be supplied using Signal Server instead of command line).

### CODE REVIEW

**Review by PerfectSlayer — APPROVED** [review]
I skimmed through the code changes and assumed it was mostly refactoring so LGTM.

File: dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/events/TestEventsHandlerImpl.java [hunk 1]
```diff
@@ -64,18 +68,22 @@ public TestEventsHandlerImpl(
...
   public void onTestModuleStart() {
-    // needed to support JVMs that run tests for multiple modules, e.g. Maven in non-forking mode
+    testModule();
+  }
+
+  private DDTestModuleImpl testModule() {
     if (testModule == null) {
-      testModule = createTestModule();
+      synchronized (this) {
+        if (testModule == null) {
```
Comment (tonyredondo) [inline dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/events/TestEventsHandlerImpl.java:81]: haha the if/lock/if trick, didn't know it was a thing in Java.

File: dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/events/TestEventsHandlerImpl.java [hunk 2]
```diff
@@ -64,18 +68,22 @@ public TestEventsHandlerImpl(
...
-    testModule = createTestModule();
   }
 
   @Override
   public void onTestModuleStart() {
-    // needed to support JVMs that run tests for multiple modules, e.g. Maven in non-forking mode
+    testModule();
+  }
+
+  private DDTestModuleImpl testModule() {
```
Comment (tonyredondo) [inline dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/events/TestEventsHandlerImpl.java:78]:
```suggestion
  private DDTestModuleImpl getTestModule() {
```

**Review by tonyredondo — APPROVED** [review]

File: dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/events/TestEventsHandlerImpl.java (hunk 1 above)
Comment (nikita-tkachenko-datadog) [reply dd-java-agent/agent-ci-visibility/src/main/java/datadog/trace/civisibility/events/TestEventsHandlerImpl.java:81]: yup, the pattern even has a name: double-checked locking

**Pull request merged**